src/validators/uuid.rs

```diff
@@ -0,0 +1,184 @@
+use pyo3::intern;
+use pyo3::prelude::*;
+use pyo3::sync::GILOnceCell;
+use pyo3::types::{PyDict, PyType};
+use uuid::Builder;
+use uuid::Uuid;
+
+use crate::build_tools::is_strict;
+use crate::errors::{ErrorType, ValError, ValResult};
+use crate::input::Input;
+use crate::recursion_guard::RecursionGuard;
+use crate::tools::SchemaDict;
+
+use super::model::create_class;
+use super::model::force_setattr;
+use super::{BuildValidator, CombinedValidator, Definitions, DefinitionsBuilder, Extra, Validator};
+
+const UUID_INT: &str = "int";
+const UUID_IS_SAFE: &str = "is_safe";
+
+static UUID_TYPE: GILOnceCell<Py<PyType>> = GILOnceCell::new();
+
+fn import_type(py: Python, module: &str, attr: &str) -> PyResult<Py<PyType>> {
+    py.import(module)?.getattr(attr)?.extract()
+}
+
+fn get_uuid_type(py: Python) -> PyResult<&PyType> {
+    Ok(UUID_TYPE
+        .get_or_init(py, || import_type(py, "uuid", "UUID").unwrap())
+        .as_ref(py))
+}
+
+#[derive(Debug, Clone, Copy)]
+enum Version {
+    UUIDv1,
+    UUIDv3,
+    UUIDv4,
+    UUIDv5,
```

Comment:
It might be nice to define the discriminators explicitly because it'll probably make the mapping back and forth with integer values compile away to noops:

```suggestion
    UUIDv1 = 1,
    UUIDv3 = 3,
    UUIDv4 = 4,
    UUIDv5 = 5,
```